src/worldstate/CMakeLists.txt

```diff
@@ -1,57 +1,97 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(worldstate)
 
+## Compile as C++11, supported in ROS Kinetic and newer
+add_compile_options(-std=c++14)
+
 ## Find catkin macros and libraries
 ## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
 ## is used, also find other catkin packages
 find_package(catkin REQUIRED COMPONENTS
   roscpp
   std_msgs
   message_generation
+  gate_detect
 )
 
 add_definitions(-std=c++14)
 
-include_directories(
-    ${catkin_INCLUDE_DIRS}
-)
-
 ################################################
 ## Declare ROS messages, services and actions ##
 ################################################
 
 ## Generate messages in the 'msg' folder
 add_message_files(
   FILES
-  state_msg.msg
+  StateMsg.msg
 )
 
 ## Generate added messages and services with any dependencies listed here
 generate_messages(
-  DEPENDENCIES
-  std_msgs
+  DEPENDENCIES std_msgs
 )
 
 ###################################
 ## catkin specific configuration ##
 ###################################
 
 catkin_package(
+   INCLUDE_DIRS include
    CATKIN_DEPENDS message_runtime
 )
 
+###########
+## Build ##
+###########
+
+include_directories(
+        ${catkin_INCLUDE_DIRS}
+        ./include
+        ./routines/include
+        ./routines/Gate/include
+        ../constants
+)
+
+add_executable(world_state_node
+    src/WorldStateNode.cpp
+    src/world_state_node.cpp
+    include/WorldStateNode.h
+        include/routines/State.h
+        src/routines/State.cpp
+        include/routines/Gate/LocatingGate.h
+        src/routines/Gate/LocatingGate.cpp
+        include/routines/Gate/AlignWithGate.h
+        src/routines/Gate/AlignWithGate.cpp
+        include/routines/Gate/PassGate.h
+        src/routines/Gate/PassGate.cpp
+
+)
```

Comment:
we should make indenting here consistent with the rest of the file and remove the extra space on the last line

Reply:
Removed!